pkg/router/controller/host_admitter_test.go

```diff
@@ -0,0 +1,399 @@
+package controller
+
+import (
+	"fmt"
+	"strings"
+	"testing"
+	"time"
+
+	kapi "k8s.io/kubernetes/pkg/api"
+	"k8s.io/kubernetes/pkg/api/unversioned"
+	"k8s.io/kubernetes/pkg/watch"
+
+	routeapi "github.com/openshift/origin/pkg/route/api"
+	"github.com/openshift/origin/pkg/route/api/validation"
+)
+
+type rejectionRecorder struct {
+	rejections map[string]string
+}
+
+func (_ rejectionRecorder) rejectionKey(route *routeapi.Route) string {
+	return route.Namespace + "-" + route.Name
+}
+
+func (r rejectionRecorder) RecordRouteRejection(route *routeapi.Route, reason, message string) {
+	r.rejections[r.rejectionKey(route)] = reason
+}
+
+func admissionChecker(denyDomain string, allowWildcards bool) RouteAdmissionFunc {
+	return func(route *routeapi.Route) error {
+		if len(route.Spec.Host) < 1 {
+			return nil
+		}
+
+		if strings.HasSuffix(route.Spec.Host, "."+denyDomain) {
+			return fmt.Errorf("host is not allowed")
+		}
+
+		_, wildcard := validation.NormalizeWildcardRoute(route)
+		if wildcard && !allowWildcards {
+			return fmt.Errorf("wildcards not admitted test")
+		}
+
+		return nil
+	}
+}
+
+func TestHostAdmit(t *testing.T) {
+	p := &fakePlugin{}
+	fn := admissionChecker("domain.blocked.test", true)
+	admitter := NewHostAdmitter(p, fn, true, LogRejections)
+	tests := []struct {
+		name   string
+		route  *routeapi.Route
+		errors bool
+	}{
+		{
+			name: "No Host",
```

Comment:
This particular function doesn't vary the namespace but even if it did it would be simpler to include namespace as a variable:

    tests := []struct {
        name      string
        routeName string
        namespace string
        host      string
        errors    bool
    }{
        {
            name:      "Blocked Host",
            routeName: "blocked",
            namespace: "host",
            host:      "host.domain.blocked.test",
            errors:    true,
        },
    }

    for _, tc := range tests {
        route := &routeapi.Route{
            ObjectMeta: kapi.ObjectMeta{
                Name:      tc.routeName
                Namespace: tc.namespace,
            },
            Spec: routeapi.RouteSpec{
                Host: tc.host,
            }
        }
        err := admitter.HandleRoute(watch.Added, route)
    }